scripts/abins/atomsdata.py

```diff
@@ -118,5 +130,44 @@ def extract(self):
         # For compatibility, regenerate the dict format on-the-fly
         return {f"atom_{i}": item for i, item in enumerate(self._data)}
 
+    class JSONableAtomData(TypedDict):
+        """JSON-friendly representation of an AtomsData entry"""
+
+        coord: List[float]
+        mass: float
+        sort: int
+        symbol: str
+
+    JSONableData = Dict[str, "AtomsData.JSONableAtomData"]
+
+    def to_dict(self) -> "AtomsData.JSONableData":
+        """Get a JSON-compatible representation of the data"""
+        data: "AtomsData.JSONableData"
+        data = {
+            f"atom_{i}": {
+                "coord": item["coord"].tolist(),
+                "mass": float(item["mass"]),
+                "sort": int(item["sort"]),
+                "symbol": str(item["symbol"]),
+            }
+            for i, item in enumerate(self._data)
+        }
+        return data
+
+    @staticmethod
+    def from_dict(data: "AtomsData.JSONableData") -> "AtomsData":
+        """Construct from JSON-compatible dictionary"""
```

Comment:
can we have the params and return for this and the above method?

Reply:
This is the curse of writing a really descriptive function signature: it becomes difficult to write a docstring that isn't completely inane. I'll see what I can do :laughing:

Reply:
I have written a bit about the param! There really isn't anything to say about the return value that isn't in the type hint.